### PR TITLE
[diopter rebuild] switch to pre-commit

### PR DESCRIPTION
**Pull request by Quarub**
... and make flake8 happy.
Similarly to the dead2 branch, we don't merge into `main` here.

The flake8 config is based on the recommendations of black and I added `W503` (line break before a binary operator) to the ignore list because it clashed with the black formatting.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+max-line-length = 88
+select = C,E,F,W,B,B950
```
Comment (thetheodor) [inline .flake8:3]: why not use the defaults?

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+extend-ignore = E203, E501, W503
```
Comment (thetheodor) [inline .flake8:4]: why `E203` and `E501`?

Comment (Quarub) [reply .flake8:4]: `E501`: disabled in favor of `B950`. Quote from link "will align more with black’s “try to respect --line-length, but don’t become crazy if you can’t”". [link](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)
`E203`: Black has a different take on what "complex expressions" are and hence has different whitespace around the slice operator than E203 would want. [link](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#slices)

File: .flake8 (hunk 1 above)
Comment (Quarub) [reply .flake8:3]: I agree with the [https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length]("don't go crazy if you can't fit into precisely 88 characters"-premise) and hence use the suggested settings there.

Comment (thetheodor) [reply .flake8:3]: the tl;dr is “it’s like highway speed limits, we won’t bother you if you overdo it by a few km/h” haha

**Pull request merged**